library/core/src/fmt/rt.rs

```diff
@@ -174,8 +174,15 @@ impl Argument<'_> {
     /// let f = format_args!("{}", "a");
     /// println!("{f}");
     /// ```
+    ///
+    /// This function should _not_ be const, to make sure we don't accept
+    /// format_args!() and panic!() with arguments in const, even when not evaluated:
+    ///
+    /// ```compile_fail,E0015
+    /// const _: () = if false { panic!("a {}", "a") };
```

Comment:
Can you add a test to `tests/ui/consts/const-eval/format.rs` as well? That seems to be where we have similar tests like this -- using a doctest is somewhat unconventional.